File: offer_46.go
package main

import (
	"fmt"
	"strconv"
)

func main() {
	fmt.Print(translateNum(12258))
}

func translateNum(num int) int {
	s := strconv.Itoa(num)
	if len(s) == 1 {
		return 1
	}
	if len(s) == 2 {
		if num > 25 {
			return 1
		}
		return 2
	}
	num_1 := 1
	num_2 := 1
	i, _ := strconv.Atoi(s[:2])
	if i <= 25 {
		num_2 = 2
	}
	for i:=2;i<len(s);i++{
		j, _ := strconv.Atoi(s[i-1:i+1])
		if s[i-1] != '0' && j <= 25{
			num_1, num_2 = num_2, num_1 + num_2
		} else {
			num_1, num_2 = num_2, num_2
		}
	}
	return num_2

}

File: offer_46.py
class Solution:
    def translateNum(self, num: int) -> int:
        _str = str(num)
        if len(_str) == 1:
            return 1
        if len(_str) == 2:
            return 1 if num > 25 else 2
        num_1 = 1
        num_2 = 1 if int(_str[:2]) > 25 else 2
        for i, _num in enumerate(_str[2:], 2):
            # 当前数字可以和上一个数字结合
            if _str[i-1] != "0" and int(_str[i - 1:i + 1]) <= 25:
                num_1, num_2 = num_2, num_2 + num_1
            else:
                num_1, num_2 = num_2, num_2
        return num_2


if __name__ == '__main__':
    s = Solution()
    print(s.translateNum(12258))
    print(s.translateNum(1))
    print(s.translateNum(11))
    print(s.translateNum(26))
    print(s.translateNum(25))
    print(s.translateNum(506))
